src/main/java/br/com/caelum/clines/api/flights/FlightController.java

```diff
@@ -39,4 +38,21 @@ FlightView show(@PathVariable Long id) {
 
         return created(uri).build();
     }
+
+    @GetMapping("/search")
+    ResponseEntity<?> search(@RequestParam(name = "date", required = false)
+                             @DateTimeFormat(pattern = "dd-MM-yyyy") LocalDate date,
+                             @RequestParam(name = "country", required = false) String country,
+                             @RequestParam(name = "state", required = false) String state,
+                             @RequestParam(name = "city", required = false) String city) {
+        LocationView location = new LocationView(country, state, city);
+
+        List<FlightView> fligthts = services.searchBy(parseToLocalDateTime(date), location);
+        return ResponseEntity.ok(fligthts);
+    }
+
+    private LocalDateTime parseToLocalDateTime(LocalDate date) {
```

Comment:
It would be nice to export this method to a `LocalDateUtil` class, so in the future we could reuse it on other classes (nonblocking comment since it could be handled as a technical debt).